login/service.go

```diff
@@ -147,6 +148,10 @@ func (keycloak *keycloakOAuthProvider) Perform(ctx *app.AuthorizeLoginContext) e
 
 func (keycloak keycloakOAuthProvider) getUser(ctx context.Context, token *oauth2.Token) (*openIDConnectUser, error) {
 	client := keycloak.config.Client(ctx, token)
+	configuration, err := config.GetConfigurationData()
```

Comment:
Pass only the required configuration option into the Controller.

If multiple options, consider creating a service level config/options object to hold that info and map from config to service config in main